homeassistant/components/sisyphus/manifest.json

```diff
@@ -2,7 +2,11 @@
   "domain": "sisyphus",
   "name": "Sisyphus",
   "documentation": "https://www.home-assistant.io/integrations/sisyphus",
-  "requirements": ["sisyphus-control==3.0"],
-  "codeowners": ["@jkeljo"],
+  "requirements": [
+    "sisyphus-control==3.1.1"
```

Comment:
The updated library has pinned socketio requirements to exact versions. We don't want libraries to pin requirements to exact versions to avoid compatibility issues with other integrations that depend on the same requirement libraries.

Please adjust the pinning in the library to a minimum version or a range with as forgiving limits as possible.

Reply:
Yeah, so this is kind of an unfortunate decision by the maintainers of `python-socketio`. There are multiple backward-incompatible versions of the Socket.IO protocol, and the maintainers of the library decided to [make correspondingly backward-incompatible versions of the library (rather than have the latest version of the library support older protocols)](https://python-socketio.readthedocs.io/en/latest/intro.html#version-compatibility). I can change the pinning to be a range, but it'll have to have a <5 in there because the Sisyphus tables use an older version of the protocol. 

I agree it's a dependency time bomb, though--as soon as somebody wants to add an integration with something that uses a newer version of the Socket.IO protocol, we'll hit conflicts. I could fork the library, but that feels like more responsibility than I really want to sign up for. Thoughts? Cross that bridge when we come to it?

Reply:
For now it's ok to add a range. If there are requirements conflicts in the future, worse case, we may be forced to disable the integration.